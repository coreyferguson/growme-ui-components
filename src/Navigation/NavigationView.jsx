import React from 'react';
import PropTypes from 'prop-types';

export default function NavigationView(props) {
  return (
    <nav>
      <ul className='navigation'>
        <div className='logo'>{logo}</div>
        <div className='applicationName'>{applicationName}</div>
      </ul>
      <ul className='contextual'>
      </ul>
    </nav>
  );
}

NavigationView.propTypes = {
  logo: PropTypes.element.isRequired,
  applicationName: PropTypes.string.isRequired,
  navigationalLinks: PropTypes.arrayOf(PropTypes.element),
  contextualLinks: PropTypes.arrayOf(PropTypes.element)
};