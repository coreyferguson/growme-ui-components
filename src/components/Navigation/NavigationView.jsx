import React from 'react';
import PropTypes from 'prop-types';

export default function NavigationView(props) {
  const navigationalLinks = props.navigationalLinks
    ? props.navigationalLinks.map((link, i) => <li key={i}>{link}</li>)
    : undefined;
  const contextualLinks = props.contextualLinks
    ? props.contextualLinks.map((link, i) => <li key={i}>{link}</li>)
    : undefined;
  return (
    <nav>
      <ul className='navigation'>
        {props.logo && <li className='logo'>{props.logo}</li>}
        {navigationalLinks}
      </ul>
      <ul className='contextual'>
        {contextualLinks}
      </ul>
    </nav>
  );
}

NavigationView.propTypes = {
  logo: PropTypes.element,
  navigationalLinks: PropTypes.arrayOf(PropTypes.element),
  contextualLinks: PropTypes.arrayOf(PropTypes.element)
};