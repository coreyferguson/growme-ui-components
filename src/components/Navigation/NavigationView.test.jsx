import React from 'react';
import { shallow } from 'enzyme';
import NavigationView from './NavigationView.jsx';
import renderer from 'react-test-renderer';

describe('NavigationView', () => {
  test('snapshot', () => {
    const tree = renderer
      .create(newView())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('all properties provided', () => {
    const wrapper = shallow(newView());
    expect(wrapper.find('ul.navigation li')).toHaveLength(2);
    expect(wrapper.find('ul.navigation li').at(0).text()).toBe('growme.fyi');
    expect(wrapper.find('ul.navigation li').at(1).text()).toBe('flash');
    expect(wrapper.find('ul.contextual li')).toHaveLength(1);
    expect(wrapper.find('ul.contextual li').at(0).text()).toBe('sign in');
  });

  test('no logo provided', () => {
    const wrapper = shallow(newView({ logo: undefined }));
    expect(wrapper.find('ul.navigation li')).toHaveLength(1);
    expect(wrapper.find('ul.navigation li').at(0).text()).toBe('flash');
  });

  test('no navigationalLinks provided', () => {
    const wrapper = shallow(newView({ navigationalLinks: undefined }));
    expect(wrapper.find('ul.navigation li')).toHaveLength(1);
    expect(wrapper.find('ul.navigation li').at(0).text()).toBe('growme.fyi');
  });

  test('no contextualLinks provided', () => {
    const wrapper = shallow(newView({ contextualLinks: undefined }));
    expect(wrapper.find('ul.contextual li')).toHaveLength(0);
  });
});

function newView(propsOverride) {
  const props = Object.assign({
    logo: <a href='#'><img src='https://assets.growme.fyi/logo_medium.jpg' />growme.fyi</a>,
    navigationalLinks: [ <a href='https://flash.growme.fyi'>flash</a> ],
    contextualLinks: [ <a href='#'>sign in</a> ],
  }, propsOverride);
  return <NavigationView {...props} />;
}