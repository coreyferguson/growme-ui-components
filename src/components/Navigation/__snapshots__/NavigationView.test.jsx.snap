// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationView snapshot 1`] = `
<nav>
  <ul
    className="navigation"
  >
    <li
      className="logo"
    >
      <a
        href="#"
      >
        <img
          src="https://assets.growme.fyi/logo_medium.jpg"
        />
        growme.fyi
      </a>
    </li>
    <li>
      <a
        href="https://flash.growme.fyi"
      >
        flash
      </a>
    </li>
  </ul>
  <ul
    className="contextual"
  >
    <li>
      <a
        href="#"
      >
        sign in
      </a>
    </li>
  </ul>
</nav>
`;
